scipy/stats/boost/tests/test_boost_stats.py

```diff
@@ -0,0 +1,77 @@
+'''Boost stats tests.'''
+
+import numpy as np
+from scipy.stats import (
+    bernoulli as scipy_bernoulli,
+    beta as scipy_beta,
+    nbinom as scipy_nbinom,
+    binom as scipy_binom,
+    ncx2 as scipy_ncx2,
+)
+from scipy.stats.boost import (
+    bernoulli as boost_bernoulli,
+    beta as boost_beta,
+    nbinom as boost_nbinom,
+    binom as boost_binom,
+    ncx2 as boost_ncx2,
+)
+
+
+def test_issue_11026():
+    p = 0.5
+    assert boost_bernoulli(p).median() == 0
+    assert boost_bernoulli(p).interval(1) == scipy_bernoulli(p).interval(1)
+    assert boost_bernoulli(p).interval(0.999) == scipy_bernoulli(p).interval(0.999)
+    assert boost_bernoulli(p).interval(1-1e-14) == scipy_bernoulli(p).interval(1-1e-14)
+
+def test_issue_12635():
+    p = 0.9999999999997369
+    a, b = 76.0, 66334470.0
+    tol = 1e-2
+    assert not scipy_beta.ppf(p, a-1, b) - scipy_beta.ppf(p, a, b) < tol
+    assert boost_beta.ppf(p, a-1, b) - boost_beta.ppf(p, a, b) < tol
+
+def test_issue_12794():
+    count_list = [10, 100, 1000]
+    p = 1e-11
+    sci = [scipy_beta.isf(p, count + 1, 100000 - count) for count in count_list]
+    boo = [boost_beta.isf(p, count + 1, 100000 - count) for count in count_list]
+    assert not all(np.diff(sci) > 0)
+    assert all(np.diff(boo) > 0)
+
+def test_issue_10317():
+    alpha, n, p = 0.9, 10, 1
+    assert scipy_nbinom.interval(alpha=alpha, n=n, p=p) != (0, 0)
+    assert boost_nbinom.interval(alpha=alpha, n=n, p=p) == (0, 0)
+
+def test_issue_11134():
+    alpha, n, p = 0.95, 10, 0
+    assert scipy_binom.interval(alpha=alpha, n=n, p=p) != (0, 0)
+    assert boost_binom.interval(alpha=alpha, n=n, p=p) == (0, 0)
+
```

Comment:
Let's add some sort of test for scipy/scipy#5122, e.g.
```python3
p = 0
n = np.random.randint(100, size=10)

x = 0
# ppf = stats.binom.ppf(x, n, p)
ppf = boost_binom.ppf(x, n, p)
assert_equal(ppf, -1)

x = np.linspace(0.01, 0.99, 10)
# ppf = stats.binom.ppf(x, n, p)
ppf = boost_binom.ppf(x, n, p)
assert_equal(ppf, 0)

x = 1
ppf = boost_binom.ppf(x, n, p)
assert_equal(ppf, n)
```
Somebody will probably want it parameterized.

Reply:
Also for scipy/scipy#1603
```python3
assert_equal(boost_binom(1000, 1e-20).ppf(0.01), 0)
```

Reply:
Also for scipy#5503. Here is some code to help:
```python3
p = 0.5
x = np.logspace(3, 14, 12)
assert_allclose(boost_binom.cdf(x, 2*x, p), 0.5, atol=1e-2)

# parametrize this tes
# x, N, p, desired CDF
[300, 1000, 3/10, 0.51559351981411995636]
[3000, 10000, 3/10, 0.50493298381929698016]
[30000, 100000, 3/10, 0.50156000591726422864]
[300000, 1000000, 3/10, 0.50049331906666960038]
[3000000, 10000000, 3/10, 0.50015600124585261196]
[30000000, 100000000, 3/10, 0.50004933192735230102]
[30010000, 100000000, 3/10, 0.98545384016570790717]
[29990000, 100000000, 3/10, 0.01455017177985268670]
[29950000, 100000000, 3/10, 5.02250963487432024943e-28]
assert_allclose(boost_binom.cdf(x, n, p), cdf_desired)

# From Wolfram Alpha: CDF[BinomialDistribution[1e12, 1e-12], 2]
assert_allclose(boost_binom.cdf(2, 10**12, 10**-12), 0.91969860292869777384)
```

Reply:
New tests added covering these cases; didn't parameterize the first test for now, will wait for feedback because I don't think these are in final form yet